bundlewrap/items/dconf.py

```diff
@@ -0,0 +1,127 @@
+from ast import literal_eval
+from json import dumps
+
+from bundlewrap.exceptions import BundleError
+from bundlewrap.items import Item
+from bundlewrap.utils.text import mark_for_translation as _
+
+
+class DconfSettingsItem(Item):
+    """
+    `dconf` settings, primarily used by GNOME applications.
+    Also known as `gsettings` sometimes.
+    """
+    BUNDLE_ATTRIBUTE_NAME = 'dconf'
+    ITEM_ATTRIBUTES = {
+        'value': None,
+        'reset': False,
+    }
+    ITEM_TYPE_NAME = 'dconf'
+
+    def __repr__(self):
+        return f'<dconf user:{self.user} path:{self.path}>'
+
+    @property
+    def user(self):
+        return self.name.split('/', 1)[0]
+
+    @property
+    def path(self):
+        return '/' + self.name.split('/', 1)[1]
+
+    def _parse_result(self, result):
+        stdout = result.stdout.decode('UTF-8').strip()
+        if stdout.isdigit():
+            return int(stdout)
+        else:
+            try:
+                return literal_eval(stdout)
+            except Exception:  # skipcq: PYL-W0703
+                return stdout
+
+    def run(self, command, **kwargs):
+        result = self.node.run(
+            f'DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/$(id -u {self.user})/bus {command}',
+            user=self.user,
+            **kwargs,
+        )
+        self._command_results.append({
+            'command': command,
+            'result': result,
+        })
+        return result
+
+    def get_auto_attrs(self, items):
+        needs = set()
+        after = set()
+        for item in items:
+            # We need to depend on the user item and the GNOME packages.
+            # To increase the chance of the setting being available, we
+            # try to run this item after all package installations have
+            # succeeded.
+            if item.ITEM_TYPE_NAME == 'user' and item.name == self.user:
+                needs.add(item.id)
+            elif item.ITEM_TYPE_NAME.startswith('pkg_') and item.name.lower().startswith('gnome'):
+                needs.add(item.id)
+            elif item.ITEM_TYPE_NAME.startswith('pkg_'):
+                after.add(item.id)
+        return {
+            'after': after,
+            'needs': needs,
+        }
+
+    def cdict(self):
+        if self.attributes['reset']:
+            return None
+
+        return {
+            'value': self.attributes['value'],
+        }
+
+    def sdict(self):
+        result = self.run(f'dconf read {self.path}', may_fail=True)
+        value = self._parse_result(result)
+
+        if not value:
```

Comment:
Need to fix this. That if errorneously catches 0 as well. We only want to catch empty strings.